### ISSUE TITLE
[BUG] check-unused-l10n doesn't find methods called via context

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#524]
* **Dart code metrics version:** 4.5.0
* **Dart sdk version:**  2.12.2 (stable)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  exclude:
    - "**/*.g.dart"
    - "**/*.gr.dart"
    - "**/*.freezed.dart"
    - "**/generated/**.dart"
    - "**/*.gform.dart"
    - "ios/**"
  strong-mode:
    implicit-casts: false
    implicit-dynamic: false
  errors:
    missing_required_param: warning
    missing_return: error
  plugins:
    - dart_code_metrics

dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 27
    source-lines-of-code: 50
    number-of-parameters: 4
    maximum-nesting-level: 5
  metrics-exclude:
    - test/**
    - e2e/**
    - ios/**
  rules:
    - always-remove-listener
    - avoid-non-null-assertion
    - avoid-returning-widgets
    - avoid-unnecessary-setstate
    - avoid-unused-parameters
    - avoid-wrapping-in-padding
    - double-literal-format
    - newline-before-return
    - no-magic-number:
        allowed: [ 0, 0.5, 1, 1.5, 2, 10, 100 ]
    - no-boolean-literal-compare
    - prefer-match-file-name:
        exclude:
          - test/**
    - prefer-single-widget-per-file
    - prefer-trailing-comma
    - no-equal-then-else
    - binary-expression-operand-order
```

</details>

**What did you do? Please include the source code example causing the issue.**

```sh
flutter pub run dart_code_metrics:metrics check-unused-l10n lib -p ^S$
```

**What did you expect to happen?**
the command consider `S.of(context).title` as used l10n

**What actually happened?**
the command doesn't consider `S.of(context).title` as used l10n

**Comment by incendial** [comment]
Available in 4.7.0 🚀

**Comment by furaiev** [comment]
Hi, 
This one doesn't work in 4.9.0
